Miglioramento delle performance

espositoandrea opened the pull request:
Drastico miglioramento delle _performance_ ottenuto tramite una _(quasi)_ totale riscrittura del codice sorgente.

espositoandrea committed:
Modifica della struttura portante del sito (1)

L'obiettivo è quello di minimizzare i tempi nel cambiamento degli argomenti mostrandoli tutti tramite iframe.

espositoandrea committed:
Rimozione ricarimento pagina inutile

espositoandrea committed:
Modifica della struttura portante del sito (2)

Apportate modifiche simili a quelle del commit 6e7e8d70c70e240dd991360c430da02a94cb0a5 alle pagine "home" degli accordi e degli strumenti.

espositoandrea committed:
Aggiunte shortcut (WIP)

espositoandrea committed:
Correzione helper

espositoandrea committed:
Riaggiunta funzionalità alla pagina del profilo

espositoandrea committed:
Corrette shortcut

FIXME: Rimuovere il tasto "ricarica"

espositoandrea committed:
Inizio conversione pagine argomenti

espositoandrea committed:
Correzione errori

espositoandrea committed:
Corrette shortcut

espositoandrea committed:
Correzione errori

Corretti errori in render.js e rimosse shortcut inutili (in quanto ridondanti)

espositoandrea committed:
Prototipazione nuovo sistema dialoghi (1)

espositoandrea committed:
Correzione errore + Reinserimento della shortcut [CTRL+BACKSPACE]

Correzione errore: la funzione changeTopic non era funzionante.

espositoandrea committed:
Rimozione di un warning

espositoandrea committed:
Inserimento dialogo di conferma uscita

espositoandrea committed:
Aggiornamento dialogo warnIfIncomplete

espositoandrea committed:
Rimozione funzioni inutili

espositoandrea committed:
Aggiunto effetto grafico suggerito da @ax3lfernus

Si è rimosso il file index.html, sostituendolo con main.html (che si è provveduto a rinominare)

espositoandrea committed:
Aggiornamento dialogo quiz (WIP)

espositoandrea committed:
Merge branch 'master' into improve-performance

espositoandrea committed:
Correzione errore grafico

espositoandrea committed:
Modificato dialogo di reset del profilo + Refactoring dialogo quiz

espositoandrea committed:
Aggiunta possibilità di ricaricare la pagina senza dover ripremere "inizia" nella landing page

espositoandrea committed:
Convertito dialogo di conferma uscita dai quiz

espositoandrea committed:
variati i collegamenti di fine quiz

Se il quiz è terminato, sarà aperta la home della sezione di partenza. Se il quiz non è terminato, sarà aperta la pagina della teoria dell'argomento che si stava studiando.

espositoandrea committed:
Convertito dialogo easter egg

Con il presente commit si è completata la conversione dei dialoghi.

espositoandrea committed:
Rimossi listener inutilizzati

espositoandrea committed:
Correzione e rimozione di warnings

espositoandrea committed:
Merge branch 'master' into improve-performance

espositoandrea committed:
Corretto errore nel link 'Esci' a fine quiz

espositoandrea committed:
Modifiche alle shortcut

- Corrette shortcut in cui era richiesta la conferma
- Aggiunta shortcut per accettare un messaggio visualizzato tramite dialogo

espositoandrea committed:
Modificate pagine pop-up

Si sono modificate le pagine pop-up adattandole al nuovo sistema di dialoghi.

Questo commit deriva dalla combinazione di 9 commit, di cui si riportano le descrizioni per avere un listato di file modificati.

- Aggiornato helper home
- Aggiornato helper accordi
- Aggiornato helper profilo
- Aggiornato helper quiz
- Aggiornato helper strumenti
- Aggiornato helper teoria
- Aggiornamento mappa
- Aggiornata pagina informazioni
- Aggiunte variazioni (automatiche) ai titoli dei dialoghi

espositoandrea committed:
Corretto errore segnalato da @ax3lfernus

espositoandrea committed:
Rimossi file inutili + separati file di terze parti

espositoandrea committed:
Migliorata la gestione del testo dei pop-up

espositoandrea committed:
Corretto errore segnalato da @ax3lfernus

Non variava la navbar al click dei bottoni nella home page.

espositoandrea committed:
Merge branch 'master' into improve-performance

espositoandrea committed:
Merge branch 'master' into improve-performance

espositoandrea committed:
Modifiche alle funzioni JS

Rimozione di funzioni inutili e accorpamento di altre.

espositoandrea committed:
Correzione dei numeri di pagina (teoria)

espositoandrea committed:
Corretto errore nel salvataggio dei quiz

espositoandrea committed:
Aggiornamento delle utilità di automatizzazione

espositoandrea committed:
Aggiornamento documentazione JavaScript

Ax3lFernus committed:
Risolti errori di "redirect"

Risolto errore di "redirect" del pulsante "Esci" nei quiz di Teoria del Piano e della Batteria.

Ax3lFernus reviewed (approved):
Errori e Bug riscontrati:
- Quando viene completato il quiz della **Teoria del Pianofote** e si tenta di verificarlo, una volta arrivato all'ultima domanda, la pressione del pulsante "Esci" mostra una pagina bianca. [_Risolto_ ]
-  Quando viene completato il quiz della **Teoria della Batteria** e si tenta di verificarlo, una volta arrivato all'ultima domanda, la pressione del pulsante "Esci" mostra la lista degli argomenti a schermo intero. [_Risolto_ ]
-  Le pagine **Informazioni** e **Aiuto** sono sproporzionate rispetto alle dimensioni dello schermo.

espositoandrea commented:
> * Le pagine **Informazioni** e **Aiuto** sono sproporzionate rispetto alle dimensioni dello schermo.

@Ax3lFernus sono spoporzionate in altezza o larghezza? Puoi pubblicare uno screenshot?

Ax3lFernus reviewed (changes requested):
- Le pagine **Informazioni** e **Aiuto** sono sproporzionate in larghezza rispetto alle dimensioni dello schermo.
- Gli avvisi del "_mancato completamento della teoria_" dovrebbero essere mostrati **meno** di frequente.
- Il testo presente all'interno dell'avviso che chiede conferma del _reset del profilo_ dovrebbe avere più margine rispetto ai bordi sinistro e destro.
- L'immagine del basso, nella sezione **Strumenti**, è sproporzionata rispetto alle immagini degli altri strumenti.

_EDIT_
Allego screenshot del problema elencato al primo punto:
![Screen](https://user-images.githubusercontent.com/21280026/56044064-b5be1680-5d3e-11e9-9ab6-2ab931a6a440.png)

espositoandrea commented:
Stando al [commento](https://github.com/F-S-C/StrumentalMente/pull/12#pullrequestreview-226105955) di @Ax3lFernus, ho stilato una lista delle attività da completare per risolvere i problemi da lui segnalati.

# TODO

- [x] Ridimensionare le pagine pop-up (automaticamente)
- [x] Ridimensionare l'immagine del basso nella sezione strumenti
- [x] Aumentare i margini nel dialogo di reset del profilo

Si noti che non è stata presa in considerazione la riduzione delle richieste di conferma di prosecuzione in quanto requisito funzionale già concordato.

## Edit 1

Commit 81d50899336a83004b3486e1a606209568d9e92b: aggiunto il ridimensionamento automatico delle pagine pop-up in base allo schermo dell'utente.

## Edit 2

Commit 2308937228e30b4c89fb021e8a903df5fa5142cf: corrette le dimensioni delle immagini nelle card (tra cui quella del basso nella sezione strumenti).

## Edit 3

Commit 9025a60710a9fde07211cc354a832476f2d4bda4: aumentato il padding nel dialogo di reset dei quiz (e in tutti gli altri dialoghi).

espositoandrea committed:
Ridimensionati i pop-up

 I pop-up sono ora ridimensionati in base allo schermo dell'utente.

espositoandrea committed:
Corrette le dimensioni delle immagini nelle card

Ax3lFernus commented:
> Stando al [commento](https://github.com/F-S-C/StrumentalMente/pull/12#pullrequestreview-226105955) di @Ax3lFernus, ho stilato una lista delle attività da completare per risolvere i problemi da lui segnalati.
> 
> # TODO
> * [x]  Ridimensionare le pagine pop-up (automaticamente)
> * [x]  Ridimensionare l'immagine del basso nella sezione strumenti
> * [ ]  Aumentare i margini nel dialogo di reset del profilo
> 
> Si noti che non è stata presa in considerazione la riduzione delle richieste di conferma di prosecuzione in quanto requisito funzionale già concordato.
> 
> ## Edit 1
> Commit [81d5089](https://github.com/F-S-C/StrumentalMente/commit/81d50899336a83004b3486e1a606209568d9e92b): aggiunto il ridimensionamento automatico delle pagine pop-up in base allo schermo dell'utente.
> 
> ## Edit 2
> Commit [2308937](https://github.com/F-S-C/StrumentalMente/commit/2308937228e30b4c89fb021e8a903df5fa5142cf): corrette le dimensioni delle immagini nelle card (tra cui quella del basso nella sezione strumenti)

- Riguardo le modifiche effettuate nell'_Edit 1_: E' presente un alert (immagino per scopi di debug) quando si aprono i pop-up. L'alert è presente nel file: `.\src\assets\js\render.js` alla riga nr._394_. 
- Riguardo le modifiche effettuate nell'_Edit 2_: Il problema della card persiste. Allego screenshot:
![screen-card](https://user-images.githubusercontent.com/21280026/56048225-51538500-5d47-11e9-8b0a-7de12e465b86.PNG)

espositoandrea committed:
Aumentato il padding nel dialogo di reset dei quiz

espositoandrea committed:
Corretti errori

Corretti errori segnalati nella PR #12 nel commento di @ax3lfernus raggiungibile al link https://github.com/F-S-C/StrumentalMente/pull/12#issuecomment-482614926.

espositoandrea commented:
> @Ax3lFernus:
> [...]
> * Riguardo le modifiche effettuate nell'_Edit 1_: E' presente un alert (immagino per scopi di debug) quando si aprono i pop-up. L'alert è presente nel file: `.\src\assets\js\render.js` alla riga nr._394_.
> * Riguardo le modifiche effettuate nell'_Edit 2_: Il problema della card persiste. Allego screenshot:
>   ![screen-card](https://user-images.githubusercontent.com/21280026/56048225-51538500-5d47-11e9-8b0a-7de12e465b86.PNG)

Gli errori segnalati sono stati corretti nel commit cab59444603a89cfe61ae21841651bfac9324049.

espositoandrea committed:
Riformattati i file CSS

Riformattati i file CSS per omogeneizzarli ai file già presenti nelle altre branch.

espositoandrea committed:
Merge branch 'master' into improve-performance

espositoandrea commented:
Si è effettuato il merge della branch `master` nella branch `improve-performance` per risolvere i conflitti di merge.

espositoandrea committed:
Rimossa funzione inutilizzata

espositoandrea committed:
Corretti errori nelle funzioni di deploy

Ax3lFernus committed:
Corretto errore di visualizzazione degli accordi quando si cambiava una slide

espositoandrea commented:
Tutte le pagine degli accordi presentano dei problemi a livello di padding: vi hanno un valore maggiore rispetto alle altre pagine del sistema. Inoltre molte pagine _settano_ delle shortcut in modo proprietario (inutilmente).

Ax3lFernus commented:
> Tutte le pagine degli accordi presentano dei problemi a livello di padding: vi hanno un valore maggiore rispetto alle altre pagine del sistema.
- C'è un errore nella visualizzazione delle figures negli accordi:
![errore_visualizzazione_immagini_accordi](https://user-images.githubusercontent.com/21280026/56052421-a8f6ee00-5d51-11e9-9aa1-af73153b7f2d.PNG)

espositoandrea committed:
Corretto errore nel cambiamento di pagina degli accordi (2)

Completamento del commit c7db3884955f9f46f2d7c492ed3c2816acb8dbf9

espositoandrea commented:
> > Tutte le pagine degli accordi presentano dei problemi a livello di padding: vi hanno un valore maggiore rispetto alle altre pagine del sistema.
> 
> * C'è un errore nella visualizzazione delle figures negli accordi:
>   ![errore_visualizzazione_immagini_accordi](https://user-images.githubusercontent.com/21280026/56052421-a8f6ee00-5d51-11e9-9aa1-af73153b7f2d.PNG)

È un problema legato a quello del padding (oltre che al Js): risolvendo il primo si risolverà _quasi_ automaticamente anche il secondo.

espositoandrea committed:
Varie correzioni

- Corretti gli errori di padding nelle pagine degli accordi
- Rimossa la definizione di shortcut inutili
- Corretto errore nella visualizzazione del pop-up delle figure

Si vedano i seguenti commenti per maggiori informazioni:
- https://github.com/F-S-C/StrumentalMente/pull/12#issuecomment-482639388
- https://github.com/F-S-C/StrumentalMente/pull/12#issuecomment-482640564
- https://github.com/F-S-C/StrumentalMente/pull/12#issuecomment-482642291

espositoandrea commented:
I problemi segnalati nei seguenti commenti:

- https://github.com/F-S-C/StrumentalMente/pull/12#issuecomment-482639388 (by @espositoandrea)
- https://github.com/F-S-C/StrumentalMente/pull/12#issuecomment-482640564 (by @Ax3lFernus)
- https://github.com/F-S-C/StrumentalMente/pull/12#issuecomment-482642291 (by @espositoandrea)

Sono stati corretti nel commit b75e86af3cbe66e4db7968307ce0d010dc499ccc.

##### Edit: Screenshots
|Prima|Dopo|
|:---:|:---:|
|<img src="https://user-images.githubusercontent.com/21280026/56052421-a8f6ee00-5d51-11e9-9aa1-af73153b7f2d.PNG" width="512px" alt="errore_visualizzazione_immagini_accordi">|<img alt="Soluzione" src="https://user-images.githubusercontent.com/23329364/56053209-d5ac0500-5d53-11e9-8d33-d3988d0e978c.png" width="512px">|

Ax3lFernus reviewed (commented):
Ulteriori errori riscontrati:
- Il pulsante "_Torna alla home_" di tutti i quiz, mostra la navbar con una pagina bianca. (Allego immagine)
![image](https://user-images.githubusercontent.com/21280026/56053274-fc6a3b80-5d53-11e9-84b4-c991fad6fae1.png)
- Nei quiz, non è possibile chiudere l'applicazione né dal "close button" né dal pulsante "Esci". 
- Dopo vari minuti di utilizzo, nella consolle appare il seguente errore: 
`(node:10608) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 maximize listeners added. Use emitter.setMaxListeners() to increase limit`

espositoandrea committed:
Corretto potenziale errore

Ridotta l'altezza massima delle immagini quando aperte in una finestra modale

espositoandrea commented:
# TODO

- [x] Correzione dei bottoni "torna alla home" dei quiz
- [x] Correzione dei bottoni per la chiusura dei quiz
- [x] Correzione dei bottoni della navbar (esempio: selezionando `Accordi` &rarr; `Chitarra` e ricliccando su `Accordi` non avviene nulla, ma si dovrebbe tornare alla lista degli strumenti)
- [x] Correzione degli stili della navbar nelle pagine dei quiz
- [x] Correzione bottoni helper nei quiz
- [x] Rimozione dello scroll nei quiz

@Ax3lFernus per il problema per cui nella console sono segnalati troppi EventEmitter da Node.js è meglio creare una issue.

#### Edit: Creazione issue

È stata creata una issue (Issue #13) per tracciare il problema del messaggio di errore.

#### Edit: Correzione

Il terzo errore della lista precedente (_correzione dei bottoni della navbar_) è stato corretto nel commit 7cdb57fbaa2d85d31f1f92d3922ce9c3eb41a87c.

#### Edit: Correzione

Il primo errore della lista precedente (_correzione dei bottoni "torna alla home" dei quiz_) è stato corretto nel commit 92ff1c96bd1195a4a7b63cf658850bc56b9b27d6.

#### Edit: Nuovo errore

Il file CSS sembra non stilare bene la navbar della pagina dei quiz:

|Corretto|Corrente _(old)_|
|:---------:|:--------:|
|<img src="https://user-images.githubusercontent.com/23329364/56061907-16634880-5d6b-11e9-9871-a7f30e65c04d.png" alt="navbar corretta" width="512px">|<img src="https://user-images.githubusercontent.com/23329364/56061749-a359d200-5d6a-11e9-8785-e89355a1f84b.png" alt="navbar corrente" width="512px">|

#### Edit: Correzione

Il secondo errore della lista precedente (_correzione dei bottoni per la chiusura dei quiz_) è stato corretto nel commit 4adced9717a620ba2100c2b4293dcfa647a73de3.

#### Edit: Correzione

Il quarto errore della lista precedente (_correzione degli stili della navbar nelle pagine dei quiz_) è stato corretto nel commit a34f97d6a228f9920cbedcabf948e0d9e16102d3.

|Corretto|Corrente _(new)_|
|:---------:|:--------:|
|<img src="https://user-images.githubusercontent.com/23329364/56061907-16634880-5d6b-11e9-9871-a7f30e65c04d.png" alt="navbar corretta" width="512px">|<img src="https://user-images.githubusercontent.com/23329364/56078580-0e96b900-5dea-11e9-8ca7-7caa68c48868.png" alt="navbar corrente" width="512px">|

#### Edit: Nuovi errori

Sono stati riscontrati i seguenti errori:

1. Non viene visualizzata la pagina di aiuto al click dell'apposito bottone
   1. Non viene visualizzata la pagina di aiuto tramite _shortcut_
2. La pagina dei quiz presenta lo scroll

#### Edit: Correzione

Il sesto errore nella lista precedente (_rimozione dello scroll nei quiz_) è stato corretto nel commit 38347b706ba54ba5313b96e152f78fcd70c07812.

#### Edit: Correzione

Il quinto errore nella lista precedente (_correzione bottoni helper nei quiz_) è stato corretto nel commit 7d858202af6b11f81c065c81993e457277b0f25a.

espositoandrea committed:
Correzione errore nei bottoni della navbar

espositoandrea committed:
Correzione errore nell'uscita dai quiz

espositoandrea committed:
Correzione bottoni uscita dei quiz

espositoandrea committed:
Corretti stili navbar quiz

espositoandrea committed:
Rimozione scroll nei quiz

espositoandrea committed:
Corretto bottone helper (quiz)

espositoandrea committed:
Rimosso warning (issue #13)

espositoandrea committed:
Merge branch 'master' into improve-performance

espositoandrea commented:
Si è effettuato il merge della branch `master` per risolvere i conflitti.

Ax3lFernus reviewed (changes requested):
Come segnalato da @prewarning il pulsante `close-button` (a.k.a. times o "x") non funziona in nessuna delle pagine.

espositoandrea committed:
Corretto errore nella chiusura dell'applicazione

espositoandrea commented:
> Come segnalato da @prewarning il pulsante `close-button` (a.k.a. times o "x") non funziona in nessuna delle pagine.

L'errore è stato corretto nel commit 4603991c2d55441ef7ebd56b4d9971be0dc56ae7.

Ax3lFernus commented:
> > Come segnalato da @prewarning il pulsante `close-button` (a.k.a. times o "x") non funziona in nessuna delle pagine.
> 
> L'errore è stato corretto nel commit [4603991](https://github.com/F-S-C/StrumentalMente/commit/4603991c2d55441ef7ebd56b4d9971be0dc56ae7).
# Errore persistente
**Attenzione**: L'errore **persiste**, nonostante il tentativo di correzione effettuato nel commit [`4603991`](https://github.com/F-S-C/StrumentalMente/commit/4603991c2d55441ef7ebd56b4d9971be0dc56ae7).

Il click del "_close button_" funziona al più una volta e nelle pagine _Landing_ e _Home_. 

Ho notato che, in più di un caso e nelle pagine sopracitate, si deve procedere a cliccare continuativamente il bottone in modo da rendere visibile il Pop-up di uscita. 

In tutte le altre pagine, come credo sia già stato _intuito_, il click del bottone non mostra nulla.

## Azioni da intraprendere in caso di mancata correzione
Data la persistenza dell'errore, intraprenderò una delle seguenti strade:
- Se l'errore continuerà a persistere, sarà mia premura aprire una **Issue**, in modo da avere il tempo per la correzione dello stesso anche successivamente al _merge_ della branch.
- Se verrà corretto, procederò ad approvare la _pull request_.

espositoandrea committed:
Corretto bug segnalato in #12 (2)

espositoandrea commented:
L'errore dovrebbe ora esser stato risolto dal commit 5affc04b6a240ba535c9ebd9793e1de631115ae2. Sembrava esser dovuto a delle chiamate mal definite della funzione `require` di _Node.js_.

ReginaZaccaria commented:
Problemi ritrovati:

- Ridurre le dimensioni delle card [HOME]
- Lo scrolling prende tutto il lato destro delle pagine arrivando fino al "close-button"
- Nella pagina "suono" [TEORIA/BASE] si attiva lo scrolling per la dimensione del testo. Stesso problema è risultato in molte altre pagine.
- Nella pagina "chiavi" e "valori e pause" [TEORIA/BASE] i simboli non sono visibili. Stesso problema è risultato nella pagina "alterazioni" [TEORIA/AVANZATA]
- Il bottone del "torna all'elenco del ..." è da ridimensionare nella parte degli ACCORDI
- Le short-cut ALT+S *, ALT+A * non funzionano.

ReginaZaccaria reviewed (changes requested):
Problemi ritrovati:

- Ridurre le dimensioni delle card [HOME]
- Lo scrolling prende tutto il lato destro delle pagine arrivando fino al "close-button"
- Nella pagina "suono" [TEORIA/BASE] si attiva lo scrolling per la dimensione del testo. Stesso problema è risultato in molte altre pagine.
- Nella pagina "chiavi" e "valori e pause" [TEORIA/BASE] i simboli non sono visibili. Stesso problema è risultato nella pagina "alterazioni" [TEORIA/AVANZATA]
- Il bottone del "torna all'elenco del ..." è da ridimensionare nella parte degli ACCORDI
- Le short-cut ALT+S *, ALT+A * non funzionano.

espositoandrea commented:
> Problemi ritrovati: [...]

Si riporta la lista di problemi segnalata da @ReginaZaccaria sotto forma di ToDo List, in modo da poter facilmente traccia dei problemi risolti da risolvere. Si editerà il seguente commento riportando man mano le modifiche effettuate.

- [x] Ridurre le dimensioni delle card [HOME]
- [x] Nella pagina "suono" [TEORIA/BASE] si attiva lo scrolling per la dimensione del testo. Stesso problema è risultato in molte altre pagine.
- [x] Il bottone del "torna all'elenco del ..." è da ridimensionare nella parte degli ACCORDI
- [x] Le short-cut ALT+S *, ALT+A * non funzionano. 
- [x] Lo scrolling prende tutto il lato destro delle pagine arrivando fino al "close-button"
- [ ] ~Nella pagina "chiavi" e "valori e pause" [TEORIA/BASE] i simboli non sono visibili. Stesso problema è risultato nella pagina "alterazioni" [TEORIA/AVANZATA]~

### Edit 16/04/2019 16:00

Corretto errore nei bottoni "Torna all'elenco [...]" nel commit 91faa201eed3a818995795dddeed8653c3200983.

### Edit 17/04/2019 12:23

_Tentativo_ di correzione dell'errore grafico delle cards nel commit e1fbf10fee0b0c67c1fcb79c226fc665c47a55ce. È necessaria una verifica su schermi a bassa risoluzione.

### Edit 17/04/2019 13:45

Aperta _issue_ #14 per tracciare la risoluzione del problema grafico dato dalla _scrollbar_.

### Edit 17/04/2019 13:52

Corretto errore nelle shortcut `ALT+S *` | `ALT+A *` nel commit acf0fdcf4a01f5be684cfad5cad68960257b9aa0.

### Edit 18/04/2019 13:51

Si è tentato di correggere l'errore riguardante la dimensione dei font nel commit 1b6ec9ee62b57620fc67b7842bb0a316089fb4aa. È necessaria una conferma della risoluzione da parte di @ReginaZaccaria prima di spuntare l'errore dalla lista dei ToDo.

### Edit 18/04/2019 13:53

Non è possibile correggere l'errore dei simboli in quanto risulta essere dipendente dalla versione di Chromium e dei font installati sulla macchina dell'utente. È quindi rimosso da questa ToDo list.

### Edit 18/04/2019 16:58

Si è corretto l'errore nella scrollbar e si è chiusa l'issue #14.

espositoandrea committed:
Correzione errore grafico (bottone indice)

Correzione dell'errore segnalato in https://github.com/F-S-C/StrumentalMente/pull/12#pullrequestreview-226487773 da @reginazaccaria riguardante i bottoni "Torna all'elenco".

espositoandrea committed:
Tentativo di correzione errore grafico cards

espositoandrea committed:
Corretto errore nelle shortcut ALT+S|A *

espositoandrea committed:
Tentativo di correzione errore "Dimensione font"

espositoandrea committed:
Modifica alle scrollbar

espositoandrea committed:
Miglioramento tasto "apri sidebar"

espositoandrea committed:
Corretto errore nelle scrollbar

Ax3lFernus reviewed (changes requested):
Il programma, una volta compilato per la distribuzione e installato, non risponde ai comandi nella _landing page_ rendendo il programma inutilizzabile (e non terminabile, dato che anche il pulsante _close-button_ non risponde).

espositoandrea committed:
Varie modifiche

- Modificate alcune informazioni (descrizione e copyright) in package.json
- Aggiunta shortcut per aprire il pannello "ispeziona" di chromium (CTRL+SHIFT+I) per questioni di Debugging

espositoandrea committed:
Correzione grafica quiz

espositoandrea committed:
Correzione errori riscontrati dopo aver fatto il build

espositoandrea committed:
Modifica al precaricamento dei video

espositoandrea committed:
Varie modifiche al JavaScript

Si sono apportate delle modifiche per rimuovere alcuni errori e rendere più "fault-tollerant" l'applicazione.

espositoandrea committed:
Vari abbellimenti grafici

Ax3lFernus reviewed (changes requested):
### Errore persistente
Una volta installata l'app, il pulsante **INIZIA**, presente nella _landing page_, non risponde ai comandi. 
Si è effettuato un tentativo per raggiungere la _home_ tramite il tasto _Invio_ ma, anch'esso, si è rivelato inutile.
Il problema non sussiste, invece, per i pulsanti "di sistema" presenti in alto a destra che, adesso, rispondono correttamente ai comandi.

espositoandrea commented:
> ### Errore persistente
> Una volta installata l'app, il pulsante **INIZIA**, presente nella _landing page_, non risponde ai comandi.
> Si è effettuato un tentativo per raggiungere la _home_ tramite il tasto _Invio_ ma, anch'esso, si è rivelato inutile.
> Il problema non sussiste, invece, per i pulsanti "di sistema" presenti in alto a destra che, adesso, rispondono correttamente ai comandi.

@Ax3lFernus puoi postare uno screenshot della console di Chromium per rendere chiaro il motivo dell'errore? Ho inserito la shortcut `CTRL+SHIFT+I` per aprire la suddetta console. Purtroppo io non riesco a ottenere questo problema per poterlo studiare e risolvere.

montanarograziano committed:
Modifica precaricamento video del pianoforte

prewarning commented:
**Errori ritrovati in data 20/04/2019**

- L'applicazione una volta aperta presenta una doppia scroll-bar nella parte destra dello schermo.
- Una volta entrati in una pagina della sezione Teoria, è impossibile ritornare nella pagina principale di Teoria, né tramite pulsante né tramite shortcut.
- Le pagine a finestra modale ( Aiuto,Mappa, Credits) presentano bug grafici:
       1. La loro titlebar scrolla in relazione al frame principale di StrumentalMente.
       2. Se si riduce il frame principale, e si aprono le due pagine, esse appariranno "tagliate".
       3. Non è possibile ridimensionarle; si può solo ridimensionare il frame principale per far rientrare le 
           due pagine all'interno di esso, ottenendo però il bug 2.
-

montanarograziano reviewed (changes requested):
**Errori ritrovati in data 20/04/2019**

- L'applicazione una volta aperta presenta una doppia scroll-bar nella parte destra dello schermo.
- Una volta entrati in una pagina della sezione Teoria, è impossibile ritornare nella pagina principale di Teoria, né tramite pulsante né tramite shortcut.
- Le pagine a finestra modale ( Aiuto,Mappa, Credits) presentano bug grafici:
       1. La loro titlebar scrolla in relazione al frame principale di StrumentalMente.
       2. Se si riduce il frame principale, e si aprono le due pagine, esse appariranno "tagliate".
       3. Non è possibile ridimensionarle; si può solo ridimensionare il frame principale per far rientrare le 
           due pagine all'interno di esso, ottenendo però il bug 2.

montanarograziano committed:
Correzione precaricamento video piano

Sostituito attributo preload precedentemente impostato su "none" con "metadata"

espositoandrea committed:
Correzione scrolling popup

Ax3lFernus reviewed (changes requested):
## Controllo della console di Chromium
In base a quanto richiesto da @espositoandrea, ho tentato di controllare la console di Chromium per verificare i problemi che riguardavano l'impossibilità di entrare nell'applicazione dal pulsante **INIZIA**.
### Errori all'apertura dell'Applicazione
Una volta aperta l'applicazione, è possibile notare (fin da subito) la mancanza dei font (non presenti nel sistema) utilizzati per lo sviluppo della stessa. Aprendo la console, è palese la loro mancanza infatti, Chromium stesso, lancia svariati errori a riguardo:
![Apertura_App](https://user-images.githubusercontent.com/21280026/56457449-706f9980-637b-11e9-96c3-60155e9f338e.PNG)
Inoltre è possibile notare la mancanza dei font anche dal pop-up di chiusura:
![Font Non presenti](https://user-images.githubusercontent.com/21280026/56457476-c7756e80-637b-11e9-8b46-74cbd6900f12.PNG)
### Impossibilità di "entrare" nell'Applicazione
Rigurado al pulsante "INIZIA", allego (come richiesto) lo screenshot dell'errore mostrato in console:
![POST_PREMO_BOTTONE](https://user-images.githubusercontent.com/21280026/56457473-b7f62580-637b-11e9-8808-877e2ac869ca.PNG)
**N.B:** Il medesimo errore si presenta quando si tenta di accedere alla _home_ tramite il pulsante _invio_.

espositoandrea committed:
Correzione script minify + correzione HTML

Si è modificato lo script minify (corregendo anche l'HTML) per risolvere il problema che portava al non funzionamento dell'applicazione successivamente all'installazione.

espositoandrea commented:
> ## Controllo della console di Chromium
> In base a quanto richiesto da @espositoandrea, ho tentato di controllare la console di Chromium per verificare i problemi che riguardavano l'impossibilità di entrare nell'applicazione dal pulsante **INIZIA**.
> 
> ### Errori all'apertura dell'Applicazione
> Una volta aperta l'applicazione, è possibile notare (fin da subito) la mancanza dei font (non presenti nel sistema) utilizzati per lo sviluppo della stessa. Aprendo la console, è palese la loro mancanza infatti, Chromium stesso, lancia svariati errori a riguardo:
> ![Apertura_App](https://user-images.githubusercontent.com/21280026/56457449-706f9980-637b-11e9-96c3-60155e9f338e.PNG)
> Inoltre è possibile notare la mancanza dei font anche dal pop-up di chiusura:
> ![Font Non presenti](https://user-images.githubusercontent.com/21280026/56457476-c7756e80-637b-11e9-8b46-74cbd6900f12.PNG)
> 
> ### Impossibilità di "entrare" nell'Applicazione
> Rigurado al pulsante "INIZIA", allego (come richiesto) lo screenshot dell'errore mostrato in console:
> ![POST_PREMO_BOTTONE](https://user-images.githubusercontent.com/21280026/56457473-b7f62580-637b-11e9-8808-877e2ac869ca.PNG)
> **N.B:** Il medesimo errore si presenta quando si tenta di accedere alla _home_ tramite il pulsante _invio_.

@Ax3lFernus  I problemi dovrebbero essere stati risolti nel commit 3be11a2dbc13a43bfabe0b5c76514297bf2f7efd.

Ax3lFernus reviewed (changes requested):
## Errore durante la creazione del file di installazione
Allego screenshot della console:
![image](https://user-images.githubusercontent.com/21280026/56457594-68b0f480-637d-11e9-9c61-82dec5272306.png)
**N.B:** Ho eseguito anche il comando `make install` prima di avviare il comando `make deploy` ma la situazione è rimasta invariata.

espositoandrea commented:
> **Errori ritrovati in data 20/04/2019**
> 
> * L'applicazione una volta aperta presenta una doppia scroll-bar nella parte destra dello schermo.
> * Una volta entrati in una pagina della sezione Teoria, è impossibile ritornare nella pagina principale di Teoria, né tramite pulsante né tramite shortcut.
> * Le pagine a finestra modale ( Aiuto,Mappa, Credits) presentano bug grafici:
>   1. La loro titlebar scrolla in relazione al frame principale di StrumentalMente.
>   2. Se si riduce il frame principale, e si aprono le due pagine, esse appariranno "tagliate".
>   3. Non è possibile ridimensionarle; si può solo ridimensionare il frame principale per far rientrare le
>   due pagine all'interno di esso, ottenendo però il bug 2.

Si è corretto l'errore dello scrolling della titlebar nel commit 2e1df0475be612fea0230e2335990dfb6f37fa1c

espositoandrea commented:
> ## Errore durante la creazione del file di installazione
> Allego screenshot della console:
> ![image](https://user-images.githubusercontent.com/21280026/56457594-68b0f480-637d-11e9-9c61-82dec5272306.png)
> **N.B:** Ho eseguito anche il comando `make install` prima di avviare il comando `make deploy` ma la situazione è rimasta invariata.

@Ax3lFernus è necessario installare lo script css_html_js_minify di python eseguendo questo comando:

```
pip install css-html-js-minify
```

Le dipendenze degli script Python non fanno parte del `Makefile`.

Ax3lFernus reviewed (changes requested):
Adesso il pulsante **INIZIA** funziona in modo corretto ma il resto dell'Applicazione non risponde ai comandi, compresi i pulsanti "di sistema", fatta eccezione per il pulsante di "Ricarica - Debug".

espositoandrea committed:
Aggiunto ridimensionamento automatico popup

espositoandrea commented:
> **Errori ritrovati in data 20/04/2019**
> 
> * L'applicazione una volta aperta presenta una doppia scroll-bar nella parte destra dello schermo.
> * Una volta entrati in una pagina della sezione Teoria, è impossibile ritornare nella pagina principale di Teoria, né tramite pulsante né tramite shortcut.
> * Le pagine a finestra modale ( Aiuto,Mappa, Credits) presentano bug grafici:
>   1. La loro titlebar scrolla in relazione al frame principale di StrumentalMente.
>   2. Se si riduce il frame principale, e si aprono le due pagine, esse appariranno "tagliate".
>   3. Non è possibile ridimensionarle; si può solo ridimensionare il frame principale per far rientrare le
>   due pagine all'interno di esso, ottenendo però il bug 2.

Con il commit e94c9d34789f39256ed88f558a3020ba1ac387c6 i popup si ridimensionano automaticamente.

espositoandrea committed:
Aggiunta possibilità di ricaricare la pagina teoria dalla navbar

espositoandrea commented:
> **Errori ritrovati in data 20/04/2019**
> 
> * L'applicazione una volta aperta presenta una doppia scroll-bar nella parte destra dello schermo.
> * Una volta entrati in una pagina della sezione Teoria, è impossibile ritornare nella pagina principale di Teoria, né tramite pulsante né tramite shortcut.
> * Le pagine a finestra modale ( Aiuto,Mappa, Credits) presentano bug grafici:
>   1. La loro titlebar scrolla in relazione al frame principale di StrumentalMente.
>   2. Se si riduce il frame principale, e si aprono le due pagine, esse appariranno "tagliate".
>   3. Non è possibile ridimensionarle; si può solo ridimensionare il frame principale per far rientrare le
>   due pagine all'interno di esso, ottenendo però il bug 2.

Si può ora (commit e67dc9afc79a8532217dc51f9cc8f29b8e0294a2) tornare all'indice della pagina "teoria" mediante la navbar.

@prewarning con questo sono terminati gli errori segnalati.

espositoandrea committed:
RImozione log di test/debug

espositoandrea committed:
Ulteriore modifica dello script di minify

espositoandrea commented:
> Adesso il pulsante **INIZIA** funziona in modo corretto ma il resto dell'Applicazione non risponde ai comandi, compresi i pulsanti "di sistema", fatta eccezione per il pulsante di "Ricarica - Debug".

@Ax3lFernus col commit 5a5d4dbf36f457ef2586815141d2184d0159b090 dovrebbe esser stato risolto l'errore (benché in modo sporco). Se le modifiche funzionano, si cercherà un _workaround_.

Potrebbe essere necessario installare lo script `jsmin` di python, tramite il seguente comando:

```
pip install jsmin
```

Ax3lFernus reviewed (changes requested):
## Problema risolto
Il problema descritto nel mio post [precedente](https://github.com/F-S-C/StrumentalMente/pull/12/#pullrequestreview-228877222) risulta corretto. Tutte le pagine funzionano a dovere.
## I font continuano a non caricarsi correttamente
I font continuano a non essere caricati. L'errore che si visualizza sulla console (mostrato nel mio post [precedente](https://github.com/F-S-C/StrumentalMente/pull/12/#pullrequestreview-228877222)) risulta invariato.

espositoandrea committed:
Correzione script minify

Correzione dello script minify.py per risolvere gli errori dei font.

espositoandrea commented:
@Ax3lFernus l'errore dei font è stato corretto nel commit e40c590ae05fffdab708d1abfd0e6ebe86bc2379. Era dovuto a una minimizzazione dei file CSS errata.

Le righe che hanno portato alla correzione dell'errore sono le seguenti ([righe `30` e `31` del file `minify.py`](https://github.com/F-S-C/StrumentalMente/blob/e40c590ae05fffdab708d1abfd0e6ebe86bc2379/minify.py#L30-L31)):

```python
data[i] = re.sub(r'"\.\./(.*?)"', r'"./\1"', data[i])
data[i] = re.sub(r'"\.\./\.\./(.*?)"', r'"../\1"', data[i])
```

Precedentemente era svolta _solo_ la seconda delle due operazioni, tramite un `if` ([vai al file](https://github.com/F-S-C/StrumentalMente/blob/5a5d4dbf36f457ef2586815141d2184d0159b090/minify.py#L30-L31) del commit 5a5d4dbf36f457ef2586815141d2184d0159b090, il precedente del commit e40c590ae05fffdab708d1abfd0e6ebe86bc2379):

```python
if "../../" in data[i]:
    data[i] = data[i].replace("../../", "../")
```

espositoandrea committed:
Modifica minify.py

espositoandrea committed:
Correzione caratteri UTF8

espositoandrea commented:
Nel commit 7f8a078e8cc5e8b7f9c37df47bbe0c5e041bbc44 si è variato lo script `minify.html` migliorando le modifiche apportate nel commit 5a5d4dbf36f457ef2586815141d2184d0159b090.

Potrebbe essere necessario dover installare `htmlmin` per Python mediante il seguente comando:

```
pip install htmlmin
```

Si è rimossa la dipendenza da `css_html_js_minify`.

espositoandrea commented:
Se non si riceve alcuna segnalazione di problemi entro l'inizio della giornata di domani (<time datetime="2019-04-21 11:00"><strong>21 Aprile 2019, ore 11:00</strong></time>), si assume che questa Pull Request sia pronta per essere chiusa mediante _merge_, in quanto stata già più volte testata.

montanarograziano reviewed (approved):
I problemi presentati risultano risolti. 
Non sono stati rilevati ulteriori problemi inerenti alla branch.

montanarograziano committed:
Rimosso bottone Debug da pagina Index

Ax3lFernus reviewed (changes requested):
## Errori riscontrati
- Nelle pagine che mostrano gli indici degli argomenti sottoforma di "macro sezioni", quando si tenta di visualizzare l'indice degli argomenti, al click su una delle sezioni, l'applicazione mostra l'animazione ma non la termina in modo corretto. Lo stesso errore è presente nelle pagine degli accordi.
Allego errore mostrato nella console: 
`Uncaught TypeError: Cannot read property 'style' of undefined
    at hideSubButtons (VM108 main.js:44)
    at showSubButtons (VM108 main.js:46)
    at window.parent.warnIfIncomplete (argomenti.html:1)
    at HTMLButtonElement.callback (VM99 render.js:3)`.
- L'accordo del piano "Do maggiore", non mostra correttamente lo schema.
- Nei quiz, quando si procede alla verifica di un argomento e ci si sposta con la tastiera, l'ultima domanda da verificare viene mostrata per una frazione di secondo facendo, di fatto, uscire l'utente dalla verifica del quiz.
- Quando si clicca il pulsante _Torna alla home_ da uno qualsiasi dei quiz, si verifica la seguente situazione:
![image](https://user-images.githubusercontent.com/21280026/56462510-953c2f00-63c4-11e9-98df-556f4cb1f5cf.png)

ReginaZaccaria commented:
Errori rilevati:

- Difficoltà nel ridurre i pop-up di grandi dimensioni
- A una precisa combinazione di comandi da tastiera, i bottoni e i tasti (avanti/indietro) non funzionano, mentre le altre short cut funzionano correttamente.

espositoandrea committed:
Revert "Rimosso bottone Debug da pagina Index"

This reverts commit b478464642072a5861b234365c2af279cb93b7ec.

espositoandrea committed:
Correzione errore shortcut segnalato da @reginazaccaria

espositoandrea committed:
Miglioramento gestione drag della finestra con i popup

espositoandrea committed:
Modifiche agli stili delle Card

espositoandrea committed:
Varie modifiche

Corretti gli errori segnalati da @ax3lfernus in https://github.com/F-S-C/StrumentalMente/pull/12#pullrequestreview-228891425 nella PR #12.

Per problemi nella minimizzazione del JavaScript e dell'HTML si è momentaneamente disattivata tale porzione dello script minify.py.

espositoandrea commented:
Nel commit 4be3922301727255c3c75ddb3852a7254ec69e6d sono stati corretti gli errori segnalati da @Ax3lFernus nel [suo commento](https://github.com/F-S-C/StrumentalMente/pull/12#pullrequestreview-228891425).

Si riporta la parte finale della descrizione del commit:

> Per problemi nella minimizzazione del JavaScript e dell'HTML si è momentaneamente disattivata tale porzione dello script `minify.py`.

espositoandrea committed:
Correzioni errori del commit 4be3922301727255c3c75ddb3852a7254ec69e6d

Ax3lFernus reviewed (changes requested):
- Il pulsante "Torna alla home" nei quiz degli accordi e nei quiz di tutti gli strumenti rimanda all'indice degli argomenti della _Teoria_.
- L'immagine dello schema dell'accordo del piano di _Do Maggiore_ non viene mostrata in modo corretto:
![DOMaj-Piano](https://user-images.githubusercontent.com/21280026/56467641-d6166100-6421-11e9-8e12-1974f2d4a8de.PNG)

espositoandrea committed:
Correzione link "Torna alla home" dei quiz

espositoandrea committed:
Correzione Makefile

espositoandrea commented:
@Ax3lFernus sono stati corretti tutti gli errori tranne quello dell'accordo di Do. Purtroppo (pur avendo testato dopo aver eseguito `make deploy`) l'errore non appare sulla mia macchina. Infatti l'accordo di Do maggiore appare in questo modo:

![image](https://user-images.githubusercontent.com/23329364/56467765-6903cb00-6423-11e9-9a0f-e8afe3f94671.png)

Se è l'unico errore riscontrabile, ignorarlo momentaneamente: se dovesse continuare a presentarsi dopo le ultime modifiche e su più macchine, allora si provvederà alla creazione di una issue.

espositoandrea commented:
Se non si riceve alcuna segnalazione di problemi entro la giornata odierna (<time datetime="2019-04-21 13:00"><strong>21 Aprile 2019, ore 13:00</strong></time>), si assume che questa Pull Request sia pronta per essere chiusa mediante _merge_, in quanto stata già più volte testata.

Ax3lFernus reviewed (changes requested):
L'errore sul redirect dei quiz si è (in parte) corretto. "Tornando alla home" dalla pagina dai quiz di uno degli strumenti si verifica il seguente errore:
![image](https://user-images.githubusercontent.com/21280026/56467916-715d0580-6425-11e9-9986-5db704143e20.png)
**N.B:** Lo stesso errore si verifica nei quiz degli accordi.

espositoandrea committed:
Correzione errore link "Torna alla home" dei quiz

espositoandrea committed:
Miglioramento UX

Revert "Correzione errore link "Torna alla home" dei quiz"

This reverts commit d474f5f32b73d5447dd5d96449ab52a8f3f2d6c6.

espositoandrea commented:
In seguito all'approvazione di tutti i membri del team @F-S-C/strumentalmente, si procede con la chiusura di questa PR tramite merge con la branch `master`.
